engines/advancedDetector.cpp

```diff
@@ -634,21 +666,68 @@ static bool getFilePropertiesIntern(uint md5Bytes, const AdvancedMetaEngine::Fil
 			return false;
 	}
 
-	if (!allFiles.contains(fname))
-		return false;
+	Common::ScopedPtr<Common::SeekableReadStream> testFile;
+
+	if (md5prop & kMD5Archive) {
+		// The desired file is inside an archive
+
+		// First, split the file string
+		Common::StringTokenizer tok(fname, ":");
+		Common::String archiveType = tok.nextToken();
+		Common::String archiveName = tok.nextToken();
+		Common::String fileName = tok.nextToken();
```

Comment:
I would do some sanity check here or better in the `AdvancedMetaEngineDetection::detectClashes()` method, so every such name contains three components.

Reply:
That's a good idea, I have added those checks to detectClashes() and can confirm they work fine.